sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SymbolicExecution/SymbolicValues/SymbolicValue.cs

```diff
@@ -330,38 +225,37 @@ protected IEnumerable<ProgramState> ThrowIfTooMany(IEnumerable<ProgramState> sta
 
             var oldBoolConstraint = oldConstraints.GetConstraintOrDefault<BoolConstraint>();
             if (oldBoolConstraint != null &&
-                oldBoolConstraint != boolConstraint)
+                oldBoolConstraint != constraint)
             {
                 return Enumerable.Empty<ProgramState>();
             }
 
             // Either same bool constraint, or previously not null, and now a bool constraint
-            return new[] { SetConstraint(boolConstraint, currentProgramState) };
+            return new[] { programState.SetConstraint(this, constraint) };
         }
 
-        private IEnumerable<ProgramState> TrySetConstraint(ObjectConstraint objectConstraint,
-            SymbolicValueConstraints oldConstraints, ProgramState currentProgramState)
+        private IEnumerable<ProgramState> TrySetObjectConstraint(ObjectConstraint constraint,

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'SonarAnalyzer.SymbolicExecution.SymbolicValueConstraint' instead of 'SonarAnalyzer.SymbolicExecution.Constraints.ObjectConstraint'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3242)